compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -1973,7 +1973,21 @@ class TypeComparer(initctx: Context) extends ConstraintHandling[AbsentContext] {
                 // instantiated to `Any` belongs to both types.
                 true
               else
-                isSameType(arg1, arg2) // TODO: handle uninstanciated types
+                isSameType(arg1, arg2) || {
+                  // We can only trust a "no" from `isSameType` when both
+                  // `arg1` and `arg2` are fully instantiated.
+                  val fullyInstantiated = new TypeAccumulator[Boolean] {
+                    override def apply(x: Boolean, t: Type) =
+                      x && {
+                        t match {
+                          case _: SkolemType | _: TypeVar | _: TypeParamRef => false
```

Comment:
It seems like there is a case missing here: I believe that inside polymorphic methods, references to TypeParameters become `TypeRef`s. I _think_ these cases can be detected by checking if `t` is a `TypeRef`